### PR TITLE
Fix bug on force plots to show feature values for specific rows in training data

### PR DESCRIPTION
**Pull request by peterataylor**
…Fixes bug where only first row feature values are show in all row plots.

### Pull Request Description

Fixes bug as described in https://github.com/alteryx/evalml/issues/3045.

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Comment by freddyaboulton** [comment]
@peterataylor Do you need help getting all the checks to pass?

**Comment by peterataylor** [comment]
> @peterataylor Do you need help getting all the checks to pass?

Thanks for the offer @freddyaboulton ! Having a quick look I should be able to do the release notes. But any guidance on the lint and dependency changes would be great. But I will look over the contrib guide again and check the approach.

**Comment by freddyaboulton** [comment]
@peterataylor The "Detect dependency changes" should be fixed as soon as you merge in the latest `main` branch.

If you have the development dependencies installed, you should be able to do the following from the top level of the repo:
```
make lint-fix
make lint
```

If `make lint` still fails, the errors should hopefully be clear. 

If you do not have make installed you can look at the Makefile to see what those commands mean and run then yourself. 

Let me know if you have any other questions!

File: evalml/model_understanding/force_plots.py [hunk 1]
```diff
@@ -45,12 +40,13 @@ def gen_force_plot(shap_values, training_data, expected_value, matplotlib):
         initjs()
 
     shap_plots = force_plot(pipeline, rows_to_explain, training_data, y)
-    for row in shap_plots:
+    for ix, row in enumerate(shap_plots):
```
Comment (freddyaboulton) [inline evalml/model_understanding/force_plots.py:43]: @peterataylor This looks good to me! Thanks for the contribution. I think the one thing that's missing is maybe a unit test to make sure we don't commit the same mistake again in the future. Maybe do one with and without a custom index on `training_data`?

Comment (peterataylor) [reply evalml/model_understanding/force_plots.py:40]: Thanks @freddyaboulton . I've added some tests to test_force_plots that use the existing parameter sets (rows 0 and 0-4). It required inspection of the returned plot object to ensure the feature values matched those that correspond to the rows_to_explain. I confirmed the tests failed with previous version and pass now. You will probably want to review though. Thanks!

**Review by freddyaboulton — APPROVED** [review]
@peterataylor This looks good to me! Thank you for the contribution.

**Pull request merged**